src/bitcoin/electrum/client.rs

```diff
@@ -57,7 +57,9 @@ impl Client {
     pub fn new(electrum_config: &config::ElectrumConfig) -> Result<Self, Error> {
         // First use a dummy config to check connectivity (no retries, short timeout).
         let dummy_config = Config::builder().retry(0).timeout(Some(3)).build();
+        // Try to ping the server.
         bdk_electrum::electrum_client::Client::from_config(&electrum_config.addr, dummy_config)
+            .and_then(|dummy_client| dummy_client.raw_call("server.ping", []))
```

Comment:
What if you just used the `ping` method?
```suggestion
            .and_then(|dummy_client| dummy_client.ping())
```